backend/src/services/package/PackageCostService.ts

```diff
@@ -0,0 +1,106 @@
+            }
+
+            if (!await S3.checkIfPackageExists(packageId)) {
+                throw new Error('404: Package not found');
+            }
+
+            // Retreive the ZIP info from S3
+            const zipBuffer = await S3.getFileByKey(packageId);
+            if(zipBuffer == null){
+                throw new Error('404: Package not found');
+            }
+
+            // Estimate the standalone cost of the package
+            this.standaloneCost =  Math.floor((zipBuffer.length *3) / 4); // Based on ratio of base64 encoding
+            return this.standaloneCost;
+        } catch (error) {
+            Logger.logError("Error fetching package cost: ", error);
+            throw error;
+        }
+    }
+
+    async getCost(packageId: string) {
+        try {
+            // Estimate the cost of the package with dependencies
+            // Setup the package URL
+            const packageUrl = await this.db.getPackageURL(packageId);
+            const urlHandler = new URLHandler(packageUrl);
+            await urlHandler.setRepoURL();
+
+            // Fetch the package.json file from the GitHub repository
+            const url = `https://api.github.com/repos/${urlHandler.getOwnerName()}/${urlHandler.getRepoName()}/contents/package.json`;
+            const response = await axios.get(url, {
+                headers: { 'Accept': 'application/vnd.github.v3.raw' }
+            });
+            const packageJson = response.data;
+
+            const dependencies = packageJson.dependencies || {};
+            const devDependencies = packageJson.devDependencies || {};
+
+            Logger.logDebug(`Dependencies: ${dependencies}`);
+            Logger.logDebug(`Dev Dependencies: ${devDependencies}`);
+
+            // Estimate the cost of the dependencies
+            let dependencyCost = 0;
+            for (const [packageName, packageVersion] of Object.entries(dependencies)) {
+                dependencyCost += await this.getDependencyCost(packageName, packageVersion as string);
+            }
+            for (const [packageName, packageVersion] of Object.entries(devDependencies)) {
+                dependencyCost += await this.getDependencyCost(packageName, packageVersion as string);
+            }
+
+            return this.standaloneCost + dependencyCost;
+        } catch (error) {
+            Logger.logError("Error fetching package cost: ", error);
+            throw error;
+        }
+    }
+
+    private cleanVersion(version: string): string{
+        return version.replace('^', '').replace('~', '');
+    }
+
+    async getDependencyCost(packageName: string, packageVersion: string) {
+        try {
+            Logger.logDebug(`Fetching dependency:  ${packageName}, ${packageVersion}`);
+
+            // Get the tar file from npm registry
+            packageVersion = this.cleanVersion(packageVersion);
+            const npmUrl = `https://registry.npmjs.org/${packageName}/${packageVersion}`;
```

Comment:
I dont think they want us to get the package dependency sizes from npmjs. From what I saw on piazza, we would calculate the dependencies from what we have in our own Package Registry. Maybe we should ask Davis or make a piazza post to clarify. You can reuse your standaloneCost func and I think it would make this part easier, and doing it recursively would be easy too. 

One of the posts asked if we do not have the package/version what do we do. And Davis said we should calculate as much as we can. So, I guess if we dont hold the dep in our Registry, we should just ignore it?? or estimate your way if possible? not sure

Additionally, Im not sure if all dependencies would be on npmjs. If a user uploads their own git repositories (without making npm pacakges on npmjs) then their package should be able to be used as a dependency in someone else's package.json? I believe.